tests/unit/recommenders/models/test_sar_singlenode.py

```diff
@@ -413,3 +418,37 @@ def test_match_similarity_type_from_json_file(header):
 
     # make sure fit still works when similarity type is loaded from a json file
     model.fit(train)
+
+
+def test_get_topk_most_similar_users(header):
+    model = SARSingleNode(**header)
+    # 1, 2, and 4 used the same items, but 1 and 2 have the same ratings also
+    train = pd.DataFrame(
+        {
+            header["col_user"]: [1, 1, 2, 2, 3, 3, 3, 3, 4, 4],
+            header["col_item"]: [1, 2, 1, 2, 3, 4, 5, 6, 1, 2],
+            header["col_rating"]: [3.0, 4.0, 3.0, 4.0, 3.0, 2.0, 1.0, 5.0, 5.0, 1.0]
+        }
+    )
+    model.fit(train)
+
+    similar_users = model.get_topk_most_similar_users(1, 1)
+    assert similar_users[header['col_user']].iloc[0] == 2
+
+    similar_users = model.get_topk_most_similar_users(2, 1)
+    assert similar_users[header['col_user']].iloc[0] == 1
+
```

Comment:
would it be possible to add another line with:
`similar_users = model.get_topk_most_similar_users(user=1, top_k=2)`
then we should check that the most similar user is 2, and then 4. 
Also, it would be good to return the prediction

Reply:
sure! for consistency I am evaluating both columns using assert_frame_equal now